### PR TITLE
(238) Advanced search form tests

### LINKED ISSUE #238: Advanced Search Form - remaining functionality
### Acceptance Criteria

```gherkin
 Scenario Outline: When user visits Advanced Search page, all page items are present
 And screen breakpoint is set to "<breakpoint>"
    Given the user navigates to "/about-cancer/treatment/clinical-trials/search/advanced"
    Then the page title is "Find NCI-Supported Clinical Trials"
    And the text "NCI-supported clinical trials are those sponsored or otherwise financially supported by NCI. See our guide, Steps to Find a Clinical Trial, to learn about options for finding trials not included in NCI's collection." appears below the title
    And "Steps to Find a Clinical Trial" link has a href "https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/trial-guide"
    And Search tip icon is displayed and text "Search Tip: All fields are optional. Skip any items that are unknown or not applicable or try our basic search." appears
    And "basic search" link has a href "/about-cancer/treatment/clinical-trials/search"
 And the following delighters are displayed
      | delighter    | href                                                       | title                               | text                                                                  |
      | cts-livehelp | /contact                                                   | Have a question? We're here to help | Chat with us:  LiveHelp Call us: 1-800-4-CANCER(1-800-422-6237)       |
      | cts-what     | /about-cancer/treatment/clinical-trials/what-are-trials    | What Are Cancer Clinical Trials?    | Learn what they are and what you should know about them.              |
      | cts-which    | /about-cancer/treatment/clinical-trials/search/trial-guide | Which trials are right for you?     | Use the checklist in our guide to gather the information you’ll need. |
   And sticky block is displayed with "Find Trials" and "Clear Form" buttons
When user scrolls to the "LeadOrganization" section
Then sticky block becomes fixed block 
   
 Examples:
            | breakpoint |
            | desktop          |
            | tablet           |
         |mobile|
```

```gherkin
Scenario: User is able to clear a form 
    Given the user navigates to "/about-cancer/treatment/clinical-trials/search/advanced"
    Then the page title is "Find NCI-Supported Clinical Trials"
    And "Age" form section is displayed
    When user types "40" in "Age" field
And user types "psa" in "Keywords" field
When user clicks on "Clear Form" button
Then "Age" input field has no value
And "Keywords" input field has no value
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by SupriyaSaxena0605**
Added tests for Advanced search form -remaining functionality.
Closes #238

### CODE REVIEW

File: cypress/integration/AdvancedSearchPage/AdvancedSearchForm/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+		cy.get('.clear-form').contains(clearbtn).should('be.visible');
+	}
+);
+
+When('user scrolls to the {string} section', (section) => {
+	cy.get(`input[aria-label="${section}"]`).scrollIntoView();
+});
+
+Then('sticky block becomes fixed block', () => {
+	cy.get('.sticky-block__anchor').should('be.visible');
```
Comment (kate-mashkina) [inline cypress/integration/AdvancedSearchPage/AdvancedSearchForm/index.js:28]: so this locator indicates an element, that is always visible
What you need to assert is that this div[class="sticky-block__anchor --sticky"] does't exist

**Review by kate-mashkina — CHANGES_REQUESTED** [review]

**Comment by kate-mashkina** [comment]
Please add the following two scenarios :

```
Scenario: user can navigate to basic search form while on advance page
		Given the user navigates to "/about-cancer/treatment/clinical-trials/search/advanced"
		Then the page title is "Find NCI-Supported Clinical Trials"
		And Search tip icon is displayed and text "Search Tip: All fields are optional. Skip any items that are unknown or not applicable or try our basic search." appears
		And "basic search" link has a href "/about-cancer/treatment/clinical-trials/search"
        When user clicks on "basic search" with href "/about-cancer/treatment/clinical-trials/search"
        Then user is redirected to "/about-cancer/treatment/clinical-trials/search"

        Scenario: user can navigate to advanced search form while on basic search form page
		Given the user navigates to "/about-cancer/treatment/clinical-trials/search"
		Then the page title is "Find NCI-Supported Clinical Trials"
		And Search tip icon is displayed and text "Search Tip: For more search options, use our advanced search." appears
        When user clicks on "advanced search" with href "/about-cancer/treatment/clinical-trials/search/advanced"
        Then user is redirected to "/about-cancer/treatment/clinical-trials/search/advanced"
```

File: cypress/integration/AdvancedSearchPage/AdvancedSearchForm/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+});
+
+When('user clicks on {string} with href {string}', (linktext, link) => {
+	cy.get('.cts-search-tip__body a').should('have.text', linktext);
+	cy.get('.cts-search-tip__body a').should('have.attr', 'href', link);
+	cy.get('.cts-search-tip__body a').click();
+});
+
+Then('user is redirected to {string}', (link) => {
+	cy.visit(link);
```
Comment (kate-mashkina) [inline cypress/integration/AdvancedSearchPage/AdvancedSearchForm/index.js:42]: cy.visit is a navigation step - what you should do is check the current path where user has already been navigated

**Review by kate-mashkina — CHANGES_REQUESTED** [review]

**Comment by arcepaul** [comment]
The git commit message is missing the # in front of 238.
`(#238) Advanced search form tests`

**Review by kate-mashkina — APPROVED** [review]

File: cypress/integration/AdvancedSearchPage/AdvancedSearchForm.feature [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		Then the page title is "Find NCI-Supported Clinical Trials"
+		And the text "NCI-supported clinical trials are those sponsored or otherwise financially supported by NCI. See our guide, Steps to Find a Clinical Trial, to learn about options for finding trials not included in NCI's collection." appears below the title
+		And "Steps to Find a Clinical Trial" link has a href "https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/trial-guide"
+		And Search tip icon is displayed and text "Search Tip: All fields are optional. Skip any items that are unknown or not applicable or try our basic search." appears
+		And "basic search" link has a href "/about-cancer/treatment/clinical-trials/search"
+		And the following delighters are displayed
+			| delighter    | href                                                       | title                              | text                                                                  |
+			| cts-livehelp | /contact                                                   | Have a question?We're here to help | Chat with us: LiveHelpCall us: 1-800-4-CANCER(1-800-422-6237)         |
```
Comment (arcepaul) [inline cypress/integration/AdvancedSearchPage/AdvancedSearchForm.feature:13]: does spacing not break with this in the title?

Comment (SupriyaSaxena0605) [reply cypress/integration/AdvancedSearchPage/AdvancedSearchForm.feature:13]: No, this is how it was being fetched.

**Pull request merged**